hamocc/mo_carchm.F90

```diff
@@ -240,11 +245,9 @@ subroutine carchm(kpie,kpje,kpke,kbnd,pdlxp,pdlyp,pddpo,prho,pglat,omask,psicomo
               ta   = ocetra(i,j,k,inatalkali) / rrho
               ah1  = nathi(i,j,k)
 
-              call CARCHM_SOLVE(s,tc,ta,sit,pt,K1,K2,Kb,Kw,Ks1,Kf,Ksi,K1p,K2p,K3p,ah1,ac,niter)
+              call CARCHM_SOLVE(psao(i,j,k),tc,ta,sit,pt,K1,K2,Kb,Kw,Ks1,Kf,Ksi,K1p,K2p,K3p,ah1,ac)
```

Comment:
Why `psao` here (and not just `s` to be consistent with the call above)?

Reply:
Yes, this didn't make sense. Fixed.